#18826 async puts

wezell opened the pull request:
Added async puts/removes

dsilvam committed:
Updating commit reference for src/main/enterprise

dsilvam committed:
Updating commit reference for src/main/enterprise

fabrizzio-dotCMS committed:
cluster initialization fails for it dependes on upgrade task (#18824)

wezell committed:
#18826 async puts

wezell committed:
#18826 async puts

dev-dotcms commented on dotCMS/src/test/java/com/dotmarketing/business/cache/provider/h22/H22CacheTest.java:
```diff
@@ -100,24 +122,75 @@ public void testInit() throws Exception {
+    
+    
+    
+    @After
```
Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms commented on dotCMS/src/test/java/com/dotmarketing/business/cache/provider/h22/H22CacheTest.java:
```diff
@@ -100,24 +122,75 @@ public void testInit() throws Exception {
+    
+    
+    @Test
+    public void test_breaking_cache_access() throws Exception {
```
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms commented on dotCMS/src/test/java/com/dotmarketing/business/cache/provider/h22/H22CacheTest.java:
```diff
@@ -100,24 +122,75 @@ public void testInit() throws Exception {
 	}
+    
+    @Test
+    public void test_dumpCacheInMiddle() throws Exception {
```
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms commented on dotCMS/src/test/java/com/dotmarketing/business/cache/provider/h22/H22CacheTest.java:
```diff
@@ -36,27 +37,37 @@
+    }
+    
+    
+    @Before
```
Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms commented on dotCMS/src/test/java/com/dotmarketing/business/cache/provider/h22/H22CacheTest.java:
```diff
@@ -100,24 +122,75 @@ public void testInit() throws Exception {
+    }
 		
+    @Test
+    public void test_multithreaded_access() throws Exception {
```
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

dev-dotcms replied:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796487)

erickgonzalez committed:
dotCMS/core#18705 PP filters bad descriptor breaks the all of the filters

* #18705 catch error if bad yaml

* #18705 IT

fmontes committed:
Updating commit reference for src/main/enterprise

victoralfaro-dotcms committed:
#18796: Adding new cicd-slack-action to workflow file (#18798)

dsilvam committed:
Revert "Updating commit reference for src/main/enterprise"

This reverts commit 3204828df3f872e5b7bab246f5d1fe48af16841a.

dsilvam committed:
Revert "Updating commit reference for src/main/enterprise"

This reverts commit 35a43ffcac1fc2a3040d229cbedf419b157a373c.

dsilvam committed:
Release 5.3.4 (#18831)

* Updating commit reference for src/main/enterprise

* Updating commit reference for src/main/enterprise

* cluster initialization fails for it dependes on upgrade task (#18824)

* Revert "Updating commit reference for src/main/enterprise"

This reverts commit 3204828df3f872e5b7bab246f5d1fe48af16841a.

* Revert "Updating commit reference for src/main/enterprise"

This reverts commit 35a43ffcac1fc2a3040d229cbedf419b157a373c.

Co-authored-by: Fabrizzio Araya <37148755+fabrizzio-dotCMS@users.noreply.github.com>

wezell committed:
Removing Travis Build Status (#18836)

victoralfaro-dotcms committed:
Enabling right name of branch when pushing to master (#18833)

wezell committed:
Issue 18744 global urlmaps (#18825)

* Update dotcmsReleaseVersion and coreWebReleasion version

* update release version

* Update dotcms-webcomponents version

* #18744 initial commit

* #18744 global urlmaps

* #18744 updating to master

* #18744 global urlmap

* #18744 tests for global urlmaps

* #18744 fixing broken tests

Co-authored-by: Freddy Montes <freddymontes@gmail.com>

fabrizzio-dotCMS committed:
#18762 apps is an enterprise feature (#18822)

* #18762 apps is an enterprise feature

* #18762 making apps an enterprise feature

* #18762 testing the invalid license scenario  from the resource

* #18762 optimize imports on test

victoralfaro-dotcms committed:
Restructuring GA config file to allow master branch execution when event is push. (#18847)

victoralfaro-dotcms committed:
Making more adjustments to GA config file to behave correctly when merging to master (#18851)

freddyDOTCMS committed:
#18187 Using whitespace analyzer with the host aliases (#18829)

* Revert release version values for master

* #18187 Using whitespace analyzer with the host aliases

* #18187 Testing

* #18187 Testing

* #18187 Testing

* #18187 Testing

* #18187 Testing

* Update HostAPITest.java

Remove set default to true

Co-authored-by: Daniel Silva <daniel.silva@dotcms.com>

dsilvam committed:
Update .gitmodules

dsilvam committed:
Update dependencies.gradle

dsilvam committed:
Update gradle.properties

erickgonzalez committed:
change coreWeb version (#18853)

victoralfaro-dotcms committed:
Fixing postman mysql step evaluation (#18852)

jcastro-dotcms committed:
#18673 : Adding validation to the objects that have a high chance of … (#18674)

* #18673 : Adding validation to the objects that have a high chance of causing NPEs because of incorrect/unexpected content information during the reindex. Improving error logging. These errors are very likely because the customer who experienced them comes from a very old version of dotCMS.

* #18673 : Adding code review changes

* #18673 testing

* #18673 Testing

Co-authored-by: freddyucv <freddy0309@gmail.com>

erickgonzalez committed:
Updating commit reference for src/main/enterprise

victoralfaro-dotcms committed:
#18524: Uptading null languages from WORKFLOW_TASK to default language (#18845)

wezell committed:
#18855 for 5.3.4 (#18861)

Commit:
Update README.md

Update links at bottom of page

wezell committed:
removing docs.

Commit:
Update README.md

Update broken links

freddyDOTCMS committed:
#18626 Changing Index.Policy to WAIT_FOR when files are upload by WebDAV (#18857)

fabrizzio-dotCMS committed:
issue #18869 fix broken reindex on auth (#18868)

jcastro-dotcms committed:
#18641 Introduce Elasticsearch scroll + more improvements for content export (#18679)

* #18641 : IMPORTANT! This is NOT the final code fix for this issue. As discussed with Will, the real code fix is a complete refactoring of this code in order to:

1. Eliminate the `EditContentletAction` class, as we've done with several other DWR classes.
2. Come up with a more efficient way of exporting high amounts of Contentlets as a CSV file.

The R&S Team needs to look into this and perform the final code changes. This PR contains the code changes that will be deployed to the customer environment in order to bring old functionality back. For this specific "temporary fix", once you have a valid functional ES Index, you need to run this curl command:
```
curl -H 'Content-Type: application/json' -XPUT 'http://localhost:9200/_all/_settings' -d '{
  "index.max_result_window" : "100000"
}'
```
The code changes involved:

- Bringing back the max result limit we had in 4.x versions. That is, 100,000 results.
- Using the Notification API for users to know (1) when the generation of the CSV file has begun, and (2) the total amount of exported content and the name of the CSV file.
- Additional logging in the `dotcms.log` file that reports the progress every 2,000 contents being added to the file.

* #18641 : Updating ES configuration values to allow dotCMS to return up to 100,000 results, and set the field limit up to 10,000 as required by several customers.

* #18641 : Read the correct properties from the es-content-settings.json file.

* #18641 : Changing notifications from the old Notification bar to the new Toast Notifications.

* #18641 performance improvements for export process

* #18641 more perf changes

* #18641 interface to exclude non-importable fields

* #18641 leave 10k window

* #18641 fixing bug found by integration-test

* #18641 cleaning up

* #18641 cleaning up

* #18641 code-review

* #18641 set proper limit

* #18641 make Unexportable an annotation instead of a flag Interface

Co-authored-by: Daniel Silva <daniel.silva@dotcms.com>

nollymar committed:
Issue 18689 move index dynamic mappings from json to code (#18751)

* #18689 Moving logic that adds ES custom mappings to a helper class

* #18689 Moving logic that adds ES custom mappings to a helper class

* #18689 New ITs

* #18689 Javadocs

* #18689 Changing logic to consider es-content.mapping.json when a  mapping is set for a field

* #18689 Removing duplicate

* #18689 Moving test to ESMappingUtilHelperTest class

* #18689 Including useful library to filter data from JSON objects

* #18689 Removing unused property

* #18689 Adding closeDBIfOpened annotation. Also, methods in the helper class were declared as instance methods

* #18689 Setting right mapping for dates and metadata

* #18689 New ITs were implemented to validate es-content-mapping.json settings are applied correctly

* #18689 Fixing failing test

* #18689 Fixing failing test

* #18689 Fixing failing test

Co-authored-by: Nollymar Longa <>

luisbacca committed:
Issue #18848 adding leading slash uri (#18850)

* #18848 : uri needs a leading slash

* #18848 : using new uri param

* #18848 : adding code review changes

Co-authored-by: Luis Bacca <luisbacca@LuisFe-MacBook-Pro.local>

fabrizzio-dotCMS committed:
Issue 18872 site-search default status fix (#18876)

* fix broken reindex on auth

* #18872 adding logic to handle sitesearch on activate and deactivate methods. This logic was previously on ESIndexResource

* #18872 ajax controller decides what api should take care of the index activation operation

dsilvam committed:
Fix Oracle and MSSQL integration tests (#18879)

victoralfaro-dotcms committed:
#18151: Upograding jackson and its dependencies to latest (2.11.1) and removing reapckaged jackson dependencies (#18864)

Commit:
Updating core-web version

dsilvam committed:
Updating commit reference for src/main/enterprise

dsilvam committed:
Removing unneeded permission code (#18842)

* test debugging

* try for unit test

* fixing test

* isMSSQL in a try

* dbconnectionfactory trying

* removing unneeded permission code

* fixed in one place

* fixed in one place

* fixing getting DYNAMIC_CONTENT_PATH

Co-authored-by: Will Ezell <will@dotcms.com>

dsilvam committed:
Revert "Updating commit reference for src/main/enterprise"

This reverts commit b2e45d45d8608388e09fdb0b630cae44eda1c938.

dsilvam committed:
Merge branch 'master' into release-5.3.4

dsilvam committed:
Merge pull request #18886 from dotCMS/release-5.3.4

Release 5.3.4

dsilvam committed:
Fixes for postman tests. Fixes #18865 (#18867)

* Fixes for postman tests. Fixes #18865

* Fix ES Index resource tests

* more fixes

* More fixes

* more fixes

* merge with master

* merge with master

wezell committed:
Issue 18870 allow es hostnames array (#18877)

* #18870 allowing multiple hosts, also ES_ENDPOINTS for a comma separated list of endpoints to try
'

* #18870 no reason to have multiple properties

* #18870 logging

dsilvam committed:
Turn on Hikari logging (#18888)

* Turn on Hikari logging

* proper defaults for Hikari

* Remove test query

dsilvam committed:
Update deploy.gradle (#18891)

wezell committed:
#18826 better hikari defaults

wezell committed:
#18826 better hikari defaults

wezell committed:
#18826 using the DotConcurrentFactory but not 100 in love- it really needs a builder pattern or something where you can specify min/max threads

wezell committed:
#18826 fixing merge
